code/graph_algorithms/test/push_relabel/test_push_relabel.cpp

```diff
@@ -0,0 +1,83 @@
+#include <iostream>
+#include <vector>
+#include "../../src/push_relabel/push_relabel.cpp"
+
+
+int g_testCounter = 0;
+std::vector<std::vector<int> > g_graph;
+std::vector<std::vector<int> > g_flow;
+
+void test(int res, int s, int t, bool empty)
+{
+	std::cout << "Test " << ++g_testCounter << ": ";
+	int k = (empty ? maxFlowEmpty(g_graph,s,t) : maxFlow(g_graph,g_flow,s,t));
+	if(k == res)
+		std::cout << "Success" << std::endl;
+	else
+		std::cout << "Fail. Expected: " << res << ", recieved: " << k << std::endl;
+	
+
+	g_graph.clear();
+	g_flow.clear();
+}
+
+int main()
+{
+	std::cout << "Testing the push-relabel algorithm" << std::endl;
+
+	g_graph.push_back(std::vector<int>{ 0, 1, 0 });
+	g_graph.push_back(std::vector<int>{ 0, 0, 1 });
+	g_graph.push_back(std::vector<int>{ 0, 0, 0 });
+
+	test(1,0,2,true);
```

Comment:
Is it possible you could use the built in testing library (Catch): https://github.com/OpenGenus/cosmos/tree/master/test/c%2B%2B?

Reply:
like this?